session/session_test.go

```diff
@@ -3965,6 +4007,59 @@ func (s *testSessionSuite) TestValidateReadOnlyInStalenessTransaction(c *C) {
 	}
 }
 
+func (s *testSessionSerialSuite) TestSpecialSQLInStalenessTxn(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	testcases := []struct {
+		name        string
+		sql         string
+		sameSession bool
+	}{
+		{
+			name:        "ddl",
+			sql:         "create table t (id int, b int,INDEX(b));",
+			sameSession: false,
+		},
+		{
+			name:        "set global session",
+			sql:         `SET GLOBAL sql_mode = 'STRICT_TRANS_TABLES,NO_AUTO_CREATE_USER';`,
+			sameSession: true,
+		},
+		{
+			name:        "analyze table",
+			sql:         "analyze table t",
+			sameSession: true,
+		},
+		{
+			name:        "session binding",
+			sql:         "CREATE SESSION BINDING FOR  SELECT * FROM t WHERE b = 123 USING SELECT * FROM t IGNORE INDEX (b) WHERE b = 123;",
+			sameSession: true,
+		},
+		{
+			name:        "global binding",
+			sql:         "CREATE GLOBAL BINDING FOR  SELECT * FROM t WHERE b = 123 USING SELECT * FROM t IGNORE INDEX (b) WHERE b = 123;",
+			sameSession: true,
+		},
+		{
+			name:        "grant statements",
+			sql:         "GRANT ALL ON test.* TO 'newuser';",
+			sameSession: false,
+		},
+		{
+			name:        "revoke statements",
+			sql:         "REVOKE ALL ON test.* FROM 'newuser';",
+			sameSession: false,
+		},
+	}
+	tk.MustExec("CREATE USER 'newuser' IDENTIFIED BY 'mypassword';")
+	for _, testcase := range testcases {
+		tk.MustExec(`START TRANSACTION READ ONLY WITH TIMESTAMP BOUND READ TIMESTAMP '2020-09-06 00:00:00';`)
+		c.Assert(tk.Se.GetSessionVars().TxnCtx.IsStaleness, Equals, true)
```

Comment:
updated.

Reply:
Eh.. I did not mean to use `c.Log`. It will print the name whatever the case may be(fail or pass).

`c.Assert(err, IsNil, Commentf("fgfg"))` will only print "fgfg" if err is not nil.

Anyway, it LGTM.